server.go

```diff
@@ -201,13 +214,16 @@ func (server *Server) handle(s *streamWrap) error {
 	}()
 
 	// Call service and respond
-	return service.svcCall(s, mtype, svcID, ctxv, argv, replyv)
+	return service.svcCall(ctx, s, mtype, svcID, ctxv, argv, replyv)
 }
 
 // svcCall calls the actual method associated
-func (s *service) svcCall(sWrap *streamWrap, mtype *methodType, svcID ServiceID, ctxv, argv, replyv reflect.Value) error {
+func (s *service) svcCall(ctx context.Context, sWrap *streamWrap, mtype *methodType, svcID ServiceID, ctxv, argv, replyv reflect.Value) error {
```

Comment:
if you carry `ctx`, then maybe we don't need to pass `ctxv` as well and we can do the `ctxv := reflect.ValueOf(ctx)` inside of this? (unless I'm missing something, seems we're essentially passing the same thing twice)